lib/autorestPlugin/pluginHost.js

```diff
@@ -0,0 +1,204 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+Object.defineProperty(exports, "__esModule", { value: true });
+const vscode_jsonrpc_1 = require("vscode-jsonrpc");
+const linq = require('linq');
+const jsonPath = require('jsonpath');
+const yaml = require("./yaml.js");
+const utils = require("../util/utils");
+const validator = require("../validate");
+const log = require('../util/logging');
+const SpecValidator = require('../validators/specValidator')
+
+const openAPIDocUrl = "https://github.com/Azure/oav";
+
+exports = module.exports;
+
+class OpenApiValidationExample {
+  Process(sessionId, initiator) {
+    return initiator.ListInputs(sessionId).then(function (swaggerFileNames) {
+      const promises = [];
+      for (const swaggerFileName of swaggerFileNames) {
+        promises.push(
+          initiator.ReadFile(sessionId, swaggerFileName).then(function (swaggerFile) {
+            const swagger = yaml.Parse(swaggerFile);
```

Comment:
just use `js-yaml`'s `safeLoad` here. The homebrew parser does way more stuff (source map generation helpers, stable order when round tripping objects like `{ b: 3, a: 2 }` vs `{ a: 2, b: 3 }`) but all that is super irrelevant here 😉